docs/getting_started.md

```diff
@@ -106,15 +109,83 @@ Providing the label name to the query component will enable the deduplication.
+
+This is especially useful for the Query node to know all endpoints to query and the time windows of each nodes, thus reducing the overhead of querying all nodes in the cluster.
+
+Given a sidecar we can have it join a gossip cluster by advertising itself to other peers within the network.
+
+```
+thanos sidecar \
+    --prometheus.url            http://localhost:9090 \
+    --tsdb.path                 /var/prometheus \
+    --gcs.bucket                example-bucket \
+    --grpc-address              0.0.0.0:19091 \            # gRPC endpoint for Store API (will be used to perform PromQL queries)
+    --http-address              0.0.0.0:19191 \            # HTTP endpoint for collecting metrics on Thanos sidecar
+    --cluster.address           0.0.0.0:19391 \            # Endpoint used to meta data about the current node
+    --cluster.advertise-address 127.0.0.1:19391 \          # Location at which the node advertise itself at to other members of the cluster
+    --cluster.peers             127.0.0.1:19391 \          # Static cluster peer where the node will get info about the cluster (repeatable)
+```
+
+With the above configuration a single node will advertise itself in the cluster and query for other members of the cluster (from itself) when you add more sidecars / components you will probably want to sent `cluset.peers` to a well known peer that will allow you to discover other peers within the cluster.
+
+When a peer advertises itself / joins a gossip cluster it sends information about all the peers it currently knows about (including itself). This information for each peer allows you to see what type of component a peer is (Source, Store, Query), the peers Store API address (used for querying) and meta data about the labels and time window the peer holds information about.
+
+Once the Peer joins the cluster it will periodically update the information it sends out with new / updated information about other peers and the time windows for the metrics that it can access.
+
+```
+thanos query \
+    --http-address              0.0.0.0:19192 \         # Endpoint for Query UI
+    --grpc-address              0.0.0.0:19092 \         # gRPC endpoint for Store API
+    --cluster.address           0.0.0.0:19591 \
+    --cluster.advertise-address 127.0.0.1:19591 \
+    --cluster.peers             127.0.0.1:19391 \       # Static cluster peer where the node will get info about the cluster
+```
+
+The Query component however does not have to utilize gossip to discover other nodes and instead can be setup to use a static list of well known addresses to query. These are repeatable so can add as many endpoint as needed. However, if you only use `store` you will not be able to discover nodes added to the cluster.
```

Comment:
` if you only use `store` you will not be able to discover nodes added to the cluster.`

Can we put this into different words?  For example:
```
When to use gossip vs static stores?
- Use gossip if you want to maintain single gossip that is able to dynamically join and remove not visible components.
- Use static store when you want to have full control of what is connected to what. It is also easier to user static store options when setting up communication with remote (cross-cluster) components e.g (sidecar in different network through some proxy)
```

Reply:
updated wording and added the example below